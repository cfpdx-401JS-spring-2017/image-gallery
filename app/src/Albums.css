.Albums {
    display: flex; 
    flex-direction: row;
    flex-wrap: wrap;
}

.newAlbum {
    height: 10vh;
    width: 10vh;
    font-size: 4em;
    text-align: center;
    color: white;
}