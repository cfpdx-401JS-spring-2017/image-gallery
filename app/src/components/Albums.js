import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class Albums extends Component {
  render() {
    return (
      <h2>Albums</h2>
    );
  }
}