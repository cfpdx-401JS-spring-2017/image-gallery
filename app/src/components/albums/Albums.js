import React, { Component } from 'react';
import {
  Route,
  Link
} from 'react-router-dom';
import AddAlbum from '../AddAlbum';
import AlbumDetail from './AlbumDetail';
import pomsAPI from '../../services/pomsAPI';

function Album({ name, url, onDelete }) {
  return (
    <div>
      <Link to={url}>{name}</Link>
      <button className="delete" onClick={onDelete}>
        ðŸ—‘
      </button>
    </div>
  );
}

export default class Albums extends Component {
  constructor(props) {
    super(props);

    this.state = {
      albums: []
    };

    this.handleAdd = this.handleAdd.bind(this);
    this.handleDeleteAlbum = this.handleDeleteAlbum.bind(this);
  }

  componentDidMount() {
    pomsAPI.getAll()
      .then(albums => this.setState({ albums }));
  }

  handleAdd(newAlbum) {
    pomsAPI.addAlbum(newAlbum)
      .then(album => {
        this.setState({
          albums: [...this.state.albums, album]
        });
      });
  }

  handleDeleteAlbum(id, index) {
    pomsAPI.deleteAlbum(id)
      .then(() => {
        const albums = this.state.albums.slice();
        albums.splice(index, 1);
        this.setState({ albums });
      });
  }

  render() {
    const { albums } = this.state;
    const { match } = this.props;

    return (
      <main>
        <h2>Albums</h2>
        <Link to={'/'}>Go Home</Link>
        <div>
          <ul>
            {albums &&
              albums.map((album, i) => <Album
                key={album._id}
                {...album}
                url={`${match.url}/${album._id}`}
                onDelete={() => this.handleDeleteAlbum(album._id, i)}
              />)}
          </ul>
          <Route path={`${match.url}/:albumId`} component={AlbumDetail} />
          <AddAlbum poms={this.state.poms} handleAdd={this.handleAdd} />
        </div>
      </main>
    );
  }
}

