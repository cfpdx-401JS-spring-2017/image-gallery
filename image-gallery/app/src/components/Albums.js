import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Album from './Album';
import AlbumContent from './AlbumContent'


export default class Albums extends Component {
    constructor(props) {
        super(props);
        this.state = {
            albums: []
        }
        this.removeImage = (albumId, imageId) => {
            fetch(`/api/albums/${albumId}/remove/${imageId}`, {
                method: 'PATCH'
            })
                .then(res => res.json())
                .then(album => {
                    console.log('album:', album);
                    const index = this.state.albums.findIndex(album => album._id === albumId)
                    const albums = this.state.albums.slice()
                    albums[index] = album
                    this.setState({ albums })
                })
                .catch(error => console.log(error));
        };

        this.addImage = (albumId, image) => {
            fetch(`/api/albums/${albumId}/images`, {
                method: 'POST',
                body: JSON.stringify(image),
                headers: new Headers({ 'Content-Type': 'application/json' })
            })
                .then(res => res.json())
                .then(album => {
                    const index = this.state.albums.findIndex(album => album._id === albumId)
                    const albums = this.state.albums.slice()
                    albums[index] = album
                    this.setState({ albums })
                })
                .catch(error => console.log(error));
        };

        this.addAlbum = (album) => {
            fetch('/api/albums', {
                method: 'POST',
                body: JSON.stringigy(album),
                headers: new Headers({ 'Content-Type': 'application/json' })
            })
                .then(res => res.json())
                .then(newAlbum => {
                    this.setState({
                        albums: [...this.state.albums, album]
                    });
                });
        }


    }

    componentDidMount() {
        const { params } = this.props.match;
        this.fetchAlbum(params.albumId)
    }

    componentWillReceiveProps(nextProps) {
        const nextAlbumId = nextProps.match.params.albumId;
        if (nextAlbumId !== this.props.match.params.albumId) {
            this.fetchStore(nextAlbumId);
        }
    }


    fetchAlbum() {
        fetch(`/api/albums`)
            .then(res => res.json())
            .then(albums => this.setState({ albums }))
            .catch(error => console.log(error));
    }

    render() {
        const { match } = this.props;
        const { albums } = this.state;
        if (!albums) return <div>Loading ...</div>;

        return (
            <div className="album-list">
                <h1>Choose an Album:</h1>
                <ul>
                    {albums.map(album => <Album
                        key={album._id} {...album}
                        url={`${match.url}/${album._id}`} />
                    )}
                </ul>

                <Route path={`${match.url}/:albumId`} render={({ match }) => {
                    const album = albums.find(album => album._id === match.params.albumId)
                    return <AlbumContent
                        {...album}
                        albumId={album._id}
                        match={match}
                        removeImage={this.removeImage}
                        addImage={this.addImage} />
                }} />
                 <h1>Add A New Album:</h1>
                 <AddAlbum
            </div>

        );

    }
}

