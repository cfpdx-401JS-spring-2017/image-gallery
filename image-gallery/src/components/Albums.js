import React, { Component } from 'react';
import albumsApi from '../albumsApi';
import { Route } from 'react-router-dom';
import Album from './Album';
import AlbumContent from './AlbumContent'


export default class Albums extends Component {
    constructor(props) {
        super(props);
        this.state = {
            albums: []
        }
    }

    componentDidMount() {
        return (albumsApi.getAlbums()
            .then((albums) => {
                this.setState({ albums })
            }))
    }

    render() {
        const { match } = this.props;
        const { albums } = this.state;
     if(!albums) return <div>Loading ...</div>;

        return (
            <div>
                <ul>
                    {albums.map(album => <Album 
                        key={album._id} {...album}
                        url={`${match.url}/${album._id}`} />
                    )}
                </ul>

                <Route path={`${match.url}/:albumId`} render={({match}) => {
                   const album= albums.find(album => album._id === match.params.albumId)
                   return <AlbumContent {...album} match={match}/>
                } }/>
            </div>

        );

    }
}

