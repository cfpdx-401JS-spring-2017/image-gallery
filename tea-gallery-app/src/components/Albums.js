import React from 'react';
import { Route, Link } from 'react-router-dom';
import AddAlbum from './AddAlbum';
import AlbumDetail from './AlbumDetail';

export default function Albums(props) {

  const { albums } = props;
  return (
    <div>
      <div className="albums-view">
        <h3>Albums</h3>
        {albums.map(album =>
          <Link key={album.name} to={`/albums/${album._id}`}>
            <button key={album.name}>
              {album.name}
            </button>
          </Link>)}
      </div>
      <Route exact path="/albums"
        render={() => {
          return <AddAlbum onAddAlbum={props.onAddAlbum} />;
        }} />
      <Route path={'/albums/:albumId'}
        render={({ match }) => {
          return <AlbumDetail
            match={match}
            onDeleteAlbum={props.onDeleteAlbum} />;
        }} />
    </div>
  );
}