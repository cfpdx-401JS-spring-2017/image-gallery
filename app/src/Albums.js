import React, { Component } from 'react';
import './Albums.css';
import Album from './Album';


class Albums extends Component {
  constructor(props) {
    super(props);
    this.state = {
      albums: [],
    }
  }

  componentDidMount() {
    fetch('/api/albums/full')
    .then(albums => albums.json())
    .then( albumsData => {
      this.setState({albums: albumsData});
    })
  }

  render() {
    let {albums} = this.state;
    return (
      <div className="Albums">
        {albums.map((album,i) => {
            return <Album key={i} albumData={album} />
        })}
        <div className="newAlbum Album">+</div>
      </div>  
    );
  }
}

export default Albums;