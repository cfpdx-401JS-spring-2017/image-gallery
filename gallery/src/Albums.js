import React, { Component } from 'react';
import './Albums.css';
import Album from './Album';
import albumsData from './data/albumsData.js';

class Albums extends Component {
  constructor(props) {
    super(props);
    this.state = {
      albums: [],
    }
    // this.albumHandler = this.albumHandler.bind(this);
  }

//   albumHandler () {
//     console.log(album);
//   }

  componentDidMount() {
    albumsData.get()
    .then(albums => {
      this.setState({albums: albums});  
    });
  }

  render() {
    let {albums} = this.state;
    return (
      <div className="Albums">
        {albums.map((album,i) => {
            return <Album key={i} albumData={album} />
        })}
      </div>  
    );
  }
}

export default Albums;