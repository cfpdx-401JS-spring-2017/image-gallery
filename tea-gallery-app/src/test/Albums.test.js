import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Albums from '../components/Albums';

const albums = [
  {
    name: 'album',
    images: [
      {
        title: 'tea leaves in teapot',
        description: 'what nice leaves'
      }
    ]
  }
];

describe('Albums', () => {
  it('renders the Albums view', () => {
    const component = renderer.create(<MemoryRouter initialEntries={['/albums', { pathname: '/albums' }]}
      initialIndex={1}>
      <Albums albums={albums} />
    </MemoryRouter>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});