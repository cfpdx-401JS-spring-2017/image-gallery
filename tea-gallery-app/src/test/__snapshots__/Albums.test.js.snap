// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums renders the Albums view 1`] = `
<div>
  <div
    className="albums-view"
  >
    <h3>
      Albums
    </h3>
    <a
      href="/albums/undefined"
      onClick={[Function]}
    >
      <button>
        album
      </button>
    </a>
  </div>
  <form
    className="add-item"
    onSubmit={[Function]}
  >
    <label>
      Add an album
      <input
        name="name"
        onChange={[Function]}
        placeholder="album name"
      />
    </label>
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
